src/main/java/edu/texas/social_computing/hospitals/HRPP.java

```diff
@@ -26,4 +33,36 @@
             // reset non-dominant partner's preference list
 
             // start from dominant partner's next hospital preference (changing the rank pointer)
+
+    /**
+     *
+     * @param residents: a list of residents in a matching that needs to be checked to see if any of the residents
+     *                 in the list violate a proximity constraint
+     * @return a set of residents that violate the proximity constraint AND are the worse placed partner
+     */
+    private Set<Resident> checkProximityViolations(List<Resident> residents) {
```

Comment:
Can we rename this to `getProximityViolations()` or `findProximityViolations()`, or something that better indicates the return value?

Reply:
yep. i was also thinking maybe `getProximityViolators()` but thought maybe that was too aggressive

Reply:
nice